dbms/src/Encryption/RateLimiter.cpp

```diff
@@ -624,17 +603,27 @@ void IORateLimiter::setStop()
 
 void IORateLimiter::runAutoTune()
 {
-    auto auto_tune_worker = [&]() {
+    auto auto_tune_and_get_io_info_worker = [&]() {
+        using time_point = std::chrono::time_point<std::chrono::system_clock>;
+        using clock = std::chrono::system_clock;
+        time_point auot_tune_time = clock::now();
+        time_point update_io_stat_time = clock::now();
         while (!stop.load(std::memory_order_relaxed))
         {
-            ::sleep(io_config.auto_tune_sec > 0 ? io_config.auto_tune_sec : 1);
-            if (io_config.auto_tune_sec > 0)
+            auto now_time_point = clock::now();
+            if ((io_config.auto_tune_sec > 0) && (now_time_point - auot_tune_time > std::chrono::seconds(io_config.auto_tune_sec)))
             {
                 autoTune();
+                auot_tune_time = now_time_point;
+            }
+            if ((bg_read_limiter || fg_read_limiter) && (now_time_point - update_io_stat_time > std::chrono::milliseconds(update_io_stat_period_ms)))
+            {
+                io_info = getCurrentIOInfo();
```

Comment:
`io_info` will be accessed concurrently.

Reply:
all fixed.